src/Syzygy.hs

```diff
@@ -124,3 +77,72 @@ interleave sigs = MkSignal $ \query -> do
   let (fromIntegral -> len) = length sigs
   (sig, n) <- zip sigs [0..]
   signal (shift (n/len) sig) query
+
+-- | Query a signal for once cycle at the given rate, relative to some absolute time
+querySignal :: forall a. Time.UTCTime -> Rational -> Interval -> Signal a -> [(Time.UTCTime, a)]
+querySignal now cps query sig = fmap formatEvent events
+  where
+    queryStart :: Rational
+    (queryStart, _) = query
+
+    events :: [Event a]
+    events = signal (pruneSignal sig) query
+
+    formatEvent :: Event a -> (Time.UTCTime, a)
+    formatEvent MkEvent{interval=(start, _), payload} =
+      let
+        delay = (start - queryStart)  * (recip cps)
+        timestamp = Time.addUTCTime (fromRational delay) now
+      in
+        (timestamp, payload)
+
+toOSCBundleTest :: (Time.UTCTime, BS.ByteString) -> BS.ByteString
+toOSCBundleTest (time, sound) = OSC.encodeOSCBundle $ OSC.OSCBundle timestamp [Right $ OSC.OSC "/play2" message]
+  where
+    timestamp :: OSC.Timestamp
+    timestamp = OSC.utcToTimestamp time
```

Comment:
I'm getting some lag, and the events are audibly a little off-time. You can fix this by adding a little latency, e.g.:

```haskell
timestamp = OSC.utcToTimestamp time `OSC.addSecs` 0.2
```